File: src/Solutions/Euler009.hs
module Solutions.Euler009 where

triples :: [[Int]]
triples = [[x, y, z] | x <- [1..444], y <- [x..444], let z = 1000 - x - y, x * x + y * y == z * z] 

euler009 :: IO String
euler009 = return $ show $ product $ head triples

File: rust/src/solutions/euler009.rs
pub fn euler009() -> String {
    for a in 3..334 {
        for b in (a + 1)..501 {
            let c = 1000 - a - b;
            if c > b && a * a + b * b == c * c {
                return (a * b * c).to_string();
            }
        }
    }
    unreachable!()
}
